### ISSUE TITLE
Emit Events when a Trigger is processed successfully

### ISSUE THREAD

**Issue by dibyom** [tektoncd/triggers#649]
# Expected Behavior

When a Trigger successfully creates a resources, we should emit a k8s event.

# Actual Behavior

Triggers does not emit any k8s events today.

# Additional Info

- We could also consider emitting events on Trigger failures -- but likely there will be many of them so this should be optional.
- Events in Pipelines: https://github.com/tektoncd/pipeline/blob/c7a60b8a77c05d8ddbc0d5a931e3f66486d512ef/docs/events.md

/kind feature

**Comment by siamaksade** [comment]
Events in case of failure are also very valuable given that the alternative is for the user to dig into the eventlistener pod logs. The use-case is that user commits a change to a GitHub repo and GitHub reports successfully invocation of the webhook but Tekton does not start the pipeline. Events would be a good starting point for finding what has gone wrong.

**Comment by dibyom** [comment]
Agreed! Though for failure events we'd also need to resolve #336 first!

**Comment by savitaashture** [comment]
@dibyom planning to move to next release
